### ISSUE TITLE
l3tl: Feature Request: Add \tl_use:nnnn with functionality like \clist_use:nnnn

### ISSUE THREAD

**Issue by mgkurtz** [latex3/latex3#1037]
As far as I see, there is currently no function in `l3tl` for transforming a token list `{Alpha}{Bravo}{Charlie}{Delta}` into `Alpha, Bravo, Charlie, and Delta` like `\clist_use:nnnn`, `\clist_use:Nnnn` and `\seq_use:Nnnn` do for `clist` and `seq`. I don’t know, whether a `\tl_use:nnnn` function misses intentionally, but I would like to see it.

Transforming the token list into a sequence first, is of course possible, but needlessly unexpandable. On the other side, turning it into a comma list just feels wrong.

So the following code puzzled together from the functions listed above and `\tl_map_function:nN` does the job:
```tex
\cs_new:Nn \tl_use:nnnn {
	\int_case:nnF { \tl_count:n {#1} } {
		{ 0 } { }
		{ 1 } { \__tl_use:nnn #1 { } { } }
		{ 2 } { \__tl_use:nnn #1 {#2} }
	} {
		\__tl_use:nwnn #1 \s__tl_stop
		\s__tl_mark
		{#3} {#4}
	}
}
\cs_new:Nn \__tl_use:nnn { \exp_not:n { #1 #3 #2 } }
\cs_new:Npn \__tl_use:nwnn #1#2 \s__tl_mark { \__tl_use:nnwnnn #2 \s__tl_mark {#1} }
\cs_new:Npn \__tl_use:nnwnnn #1#2#3 \s__tl_mark #4 #5 #6 {
	\__tl_use_none_delimit_by_s_stop:w #2
	\__tl_use_finalone:nwnnn { #4 #6 #1 } \s__tl_stop
	\__tl_use:nnwnnn {#2} #3 \s__tl_mark { #4 #5 #1 } {#5} {#6}
}
\cs_new:Npn \__tl_use_finalone:nwnnn #1#2 \s__tl_mark #3 #4 #5 { \exp_not:n {#1} }

\cs_new:Nn \tl_use:nn { \tl_use:nnnn {#1} {#2} {#2} {#2} }
\cs_generate_variant:Nn \tl_use:nnnn { V }
\cs_new_eq:NN \tl_use:Nnnn \tl_use:Vnnn
\cs_new:Nn \tl_use:Nn { \tl_use:Nnnn {#1} {#2} {#2} {#2} }
```

Are there any opinions regarding such an addition (with possibly different code)?

NB: Allowing to create a sequence expandably without saving it, would of course also solve my problem. If there where such hypothetical functions, the following much shorter code could also work:
```tex
\cs_new:Npn \tl_use:nnnn #1 { \seq_use:ennn { \seq_from_split:nn { } {#1} } }
```

**Comment by u-fischer** [comment]
Why is your data in a tl-var and not in a sequence or a comma list?

**Comment by mgkurtz** [comment]
> Why is your data in a tl-var and not in a sequence or a comma list?

TLDR: It is no tl-var, but a token list consisting of control sequences given by the user.

Good question. Perhaps I should have written that first. Actually I want to provide a user level macro usable as `\peopleentertext{ \Anne \Bert \Charlie \Donald }` yielding output like `Anne Adams, Bert Brown, Charlie Chaplin, and Donald Duck enter the meeting.` for use in minutes.

The interface has been `\peopleentertext{Anne, Bert, Charlie, Donald}` before and the command sequences where generated using good old `\csname`. But as everywhere else in the document the name macros should be used (for generating errors for misspelled names instead of typos) some users (including me) disliked the inconsistency. As we are [currently porting](https://github.com/fsmathe/protkl) our minutes template to LaTeX3, I thus changed the interface to take a token list (where all tokens happen to be control sequences as depicted above). Maybe there would be wiser choices. I would also be happy with another solution. But for the interface we wanted, we needed the functionality of `\clist_use:nnnn` but for token lists.

Of course I would not _need_ this function to be expandable, as usually it should produce typeset output.
But when I want to log this to a file (and I wanted so, for tests using the l3build framework), the expandable version just works, where the non-expandable one is not so nice.

Does this answer the question?

**Comment by josephwright** [comment]
@mgkurtz I don't think we've looked at this one yet - there's no reason to close it

**Comment by mgkurtz** [comment]
Okay, thank you for the response. Then I will it out, when I have some time left.

**Comment by FrankMittelbach** [comment]
As Joseph said we didn't have time to look at it and discuss it internally, but my vote tends to go towards *not* providing it.   Token lists are for programming not for text --- already clists are a questionable boundary case. There will be a "text" module one day where it makes much more sense imho. So for now I suggest just leave it parked and don't spend time on it before we have made up our mind to accept or reject the idea.
